en/services/command.md

```diff
@@ -76,4 +78,14 @@ sequenceDiagram;
 -->
 
 The rate at which progress messages are emitted is system-dependent.
-Generally though, the GCS should have a much increased timeout after receiving an ACK with `MAV_RESULT_IN_PROGRESS`. 
+Generally though, the GCS should have a much increased timeout after receiving an ACK with `MAV_RESULT_IN_PROGRESS`.
+
+Typically if a system receives exactly the same command while it is already processing the command, it should continue the operation.
+If the command is a "resend" (confirmation number > 0) the command can be ignored; otherwise the system should respond immediately with a progress update `COMMAND_ACK`.
```

Comment:
What about `COMMAND_INT` which don't have `confirmation`?

Generally what happens if the same command arrives twice is that the second one will be returned as "BUSY" because e.g. the autopilot is already busy with the first command.

Reply:
Thanks @julianoes !.
1. I can remove the bit about confirmation - e.g. "If the command is a "resend the ... "
2. Re Busy, presume you mean an ack of [MAV_RESULT_TEMPORARILY_REJECTED](https://mavlink.io/en/messages/common.html#MAV_RESULT_TEMPORARILY_REJECTED)? Are you suggesting that this is what we should say above or are you saying this is "an option"? I kind of prefer my suggestion for this case because it is more like "we're already doing it" than " _something_ is going on that is preventing us accepting this message". I'm OK with either approach as long as we document it clearly.

Reply:
I think `MAV_RESULT_TEMPORARILY_REJECTED` is what we're doing for many cases already.

The confirmation flag would make sense, if only we could add it to `COMMAND_INT` and if we were actually using it. I don't think we are.

Reply:
@julianoes Thanks

> The confirmation flag would make sense, if only we could add it to COMMAND_INT and if we were actually using it. I don't think we are.

FWIW the `COMMAND_INT` has a `continue` field that I "think" is supposed to do what confimation does and indicate a resend  - but it is just binary value rather than a counter. 

What I have done though is deleted this bit. It shouldn't be here because this is generic behaviour.

> I think MAV_RESULT_TEMPORARILY_REJECTED is what we're doing for many cases already.

Can you give me an example of an interruptible long running operation that does this? 

I think we need to support update commands and we can treat a resend message as the same thing - ie for example consider we have a message to send vehicle to a certain altitude that sends progress until it reaches that altitude. We should be able to send a new message that gives a new altitude without having to be rejected or wait for the same thing. 

A bigger concern for me is that I don't understand what assumptions we are making about parallelism here. Can we have a slack chat later to untangle me?  
[Specifically consider  someone has a separate cancel command. The response to this is ACK or accepted. But does the drone also need to continue to ACK the cancellation of the original message.]

Reply:
> I think we need to support update commands and we can treat a resend message as the same thing - ie for example consider we have a message to send vehicle to a certain altitude that sends progress until it reaches that altitude.

That would make sense, however, that's not how it's used right now. These things are not implemented as long running but it only starts the process to do a takeoff, and I just don't know if that would potentially change.

I agree, we probably need to have a conversation about commands in general in the next MAVLink call.

Reply:
If you're wondering about parallelism, this is how it's done in MAVSDK:
https://github.com/mavlink/MAVSDK/blob/dff24bade3760e13d81ff2be899de3b842602ef8/src/core/mavlink_commands.cpp#L8-L15
(tldr: not parallel)

I don't know how it's done for QGC.